### DISCUSSION TITLE
Memory consumption of runners in Kubernetes mode

### DISCUSSION THREAD

**Discussion by DenisPalnitsky** [actions/actions-runner-controller#3374] [General]
I'm load testing Kubernetes mode 240-360 jobs running simultaneously with a target to support ~1000 jobs. 
Currently two pods are being created for one job "runner" and "runner-worflow". Workflow pod does all the job and runner is only responsible for "managing" the workflow pod. I was surprised to learn that runner pod could reach up to 1Gb which seems a bit excessive considering it's role.
<img width="774" alt="image" src="https://github.com/actions/actions-runner-controller/assets/154290/19ac2385-26cd-4ee4-8fff-36fdde273eaa">

I can speculate that it's due to Node that is used for running hooks so I wonder if rewriting hooks to go would make it more efficient.

**Comment by mikemarquez** [comment]
Is this still relevant?  I am seeing similar resource usage in our system as well.
